libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -435,31 +436,28 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 	m_context->functionExitPoints.push(
 		CodeTransformContext::JumpInfo{m_assembly.newLabelId(), m_assembly.stackHeight()}
 	);
-	try
-	{
-		CodeTransform(
-			m_assembly,
-			m_info,
-			_function.body,
-			m_allowStackOpt,
-			m_dialect,
-			m_builtinContext,
-			m_evm15,
-			m_identifierAccess,
-			m_useNamedLabelsForFunctions,
-			m_context
-		)(_function.body);
-	}
-	catch (StackTooDeepError const& _error)
+	CodeTransform subTransform(
+		m_assembly,
+		m_info,
+		_function.body,
+		m_allowStackOpt,
+		m_dialect,
+		m_builtinContext,
+		m_evm15,
+		m_identifierAccess,
+		m_useNamedLabelsForFunctions,
+		m_context
+	);
+	subTransform(_function.body);
+	if (!subTransform.m_stackErrors.empty())
 	{
-		// This exception will be re-thrown after the end of the surrounding block.
-		// It enables us to see which functions compiled successfully and which did not.
-		// Even if we emit actual code, add an illegal instruction to make sure that tests
-		// will catch it.
-		StackTooDeepError error(_error);
-		if (error.functionName.empty())
-			error.functionName = _function.name;
-		stackError(std::move(error), static_cast<int>(height));
+		m_assembly.markAsInvalid();
+		for (auto& stackError: subTransform.m_stackErrors)
```

Comment:
```suggestion
		for (StackTooDeepError& stackError: subTransform.m_stackErrors)
```

Reply:
I'm changing it, but... :-D - does this **really** help? I mean it's clear that it's a stack error from its name - and whether I name it or not, I still have to look up the definition of the struct to find out what this actually is... so then I might as well just use ``auto`` and look up ``m_stackErrors`` as well...

Reply:
Yes, it helps! You can click "follow" on the name of the type and don't have to go through `m_stackErrors` first.